File: packages/stera_recorder/ios/stera_recorder/Sources/stera_recorder/metrics/RecordingMetricsCollector.swift
import Foundation

struct TimestampWindow {
    let minNs: Int64
    let maxNs: Int64
    let count: Int64
}

struct AssociationStats {
    let associated: Int
    let unmatched: Int
    let maxDeltaNs: Int64
}

protocol RecordingMetricsCollector {
    func computeDepthAssociations(
        rgbTimestamps: [Int64],
        depthTimestamps: [Int64],
        thresholdNs: Int64
    ) -> AssociationStats

    func computeStreamOverlap(ranges: [TimestampWindow]) -> Bool
}

File: packages/stera_recorder/android/src/main/kotlin/open/fpvlabs/stera/ar_recorder/metrics/RecordingMetricsCollector.kt
package open.fpvlabs.stera.ar_recorder.metrics

data class TimestampWindow(
    val minNs: Long,
    val maxNs: Long,
    val count: Long
)

data class AssociationStats(
    val associated: Int,
    val unmatched: Int,
    val maxDeltaNs: Long
)

interface RecordingMetricsCollector {
    fun computeDepthAssociations(
        rgbTimestamps: List<Long>,
        depthTimestamps: List<Long>,
        thresholdNs: Long
    ): AssociationStats

    fun computeStreamOverlap(ranges: List<TimestampWindow>): Boolean
}
